src/error.rs

```diff
@@ -43,6 +47,13 @@ pub enum Error {
 }
 
 impl Error {
+    pub fn context(context: impl Into<String>, error: impl Into<Box<Error>>) -> Self {
+        Error::Context {
+            context: context.into(),
+            error: error.into(),
+        }
+    }
+
```

Comment:
Not that important, but I think an associated builder-like function could be pretty ergonomic:
```suggestion
    pub fn context(self, context: impl Into<String>) -> Self {
        let error = Box::new(self);
        Error::Context {
            context: context.into(),
            error: error,
        }
    }

```

Reply:
That's much better, thanks!